### PR TITLE
bootkube: Add info to /etc/motd and a bit to /root/.bash_history

### PR DESCRIPTION
**Pull request by cgwalters**
The intention here is to help people debug.  If you like this
I may also try to do something similar for masters.

### CODE REVIEW

**Comment by crawford** [comment]
I like it.

/approve

**Comment by abhinavdahiya** [comment]
since we want this. this should not be part of the `bootkube.sh`.

@cgwalters can you add a separate unit that does this.

**Comment by crawford** [comment]
It doesn't even need to be a unit. You can write the files directly with Ignition.

**Comment by cgwalters** [comment]
OK, I updated this; using Ignition makes sense, but the stumbling block I hit is that for `/etc/motd` we need to both append and make the file world-readable.

**Comment by cgwalters** [comment]
/hold cancel

OK I read more of the code base; this doesn't really need adding a yaml file.

**Comment by wking** [comment]
I think we want to stick with the `data.Assets` approach for things like this.  I'm less worried about appending to the `motd` vs. just clobbering it, although we could special-case an append there if folks have strong feelings.  Also, I think we want the history in `/home/core/.bash_history`, where folks will log in, since you don't need elevated permissions to view the logs on the current RHCOS.  I've pushed `git://github.com/wking/openshift-installer.git motd-fixup` (b4a7f670ca64ee60f3aa33d800e88a3c09681269) with a fixup if you want to pull that in and squash onto your commit.  Thoughts?

**Comment by cgwalters** [comment]
I don't like the idea of clobbering it, we have some useful information there right now and I'd like to add more (e.g. pointing people at `toolbox`).

**Comment by cgwalters** [comment]
(Though I do agree with the `/home/core` change, and also it's true we don't need append semantics there since the user can't have logged in before)

**Comment by wking** [comment]
> However, as is in your PR won't the `bash_history` be owned by uid `0`?

Nope, because of [this][1].  You'd want something similar to set `Append` for the `motd`.  Probably worth a:

```go
filename := path.Base(uri)
```

earlier in that unpacker so we don't have to call `Base` multiple times on the same URI.

[1]: https://github.com/openshift/installer/commit/b4a7f670ca64ee60f3aa33d800e88a3c09681269#diff-2d9b655c8052ca2e3bdd9004778c75bcR203

**Comment by cgwalters** [comment]
> Nope, because of this. You'd want something similar to set Append for the motd.

Makes sense, thanks.  Applied all suggestions :arrow_up:

File: pkg/asset/ignition/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -191,13 +191,26 @@ func (a *Bootstrap) addStorageFiles(base string, uri string, templateData *boots
...
 	} else {
 		mode = 0600
 	}
 	ign := ignition.FileFromBytes(strings.TrimSuffix(base, ".template"), mode, data)
+	if filename == ".bash_history" {
+		ign.User = &igntypes.NodeUser{Name: "core"}
+		ign.Group = &igntypes.NodeGroup{Name: "core"}
+	}
+	if appendToFile {
+		ign.Append = true
```
Comment (wking) [inline pkg/asset/ignition/bootstrap/bootstrap.go]: This could just be `ign.Append = appendToFile` without the `if` wrapper, right?

Comment (cgwalters) [reply pkg/asset/ignition/bootstrap/bootstrap.go]: Sure, done :arrow_down:

**Pull request merged**